test/extended/util/client.go

```diff
@@ -265,46 +240,75 @@ func (c *CLI) SetManagedNamespace() *CLI {
-func (c CLI) WithoutNamespace() *CLI {
+func (c *CLI) WithoutNamespace() *CLI {
 	c.withoutNamespace = true
-	return &c
+	return c
 }
 
 // WithToken instructs the command should be invoked with --token rather than --kubeconfig flag
-func (c CLI) WithToken(token string) *CLI {
-	c.configPath = ""
+func (c *CLI) WithToken(token string) *CLI {
+	c.currentConfigPath = ""
 	c.token = token
-	return &c
+	return c
 }
 
-// SetupProject creates a new project and assign a random user to the project.
+// NewProject creates a new project and assign a random user to the project.
 // All resources will be then created within this project.
 // Returns the name of the new project.
-func (c *CLI) SetupProject() string {
-	exist, err := DoesApiResourceExist(c.AdminConfig(), "projects", "project.openshift.io")
+func (c *CLI) NewProject() string {
+	projectExist, err := DoesApiResourceExist(c.AdminConfig(), "projects", "project.openshift.io")
 	o.Expect(err).ToNot(o.HaveOccurred())
-	if exist {
-		return c.setupProject()
-	}
-	return c.setupNamespace()
+	return c.setupProjectOrNamespace(projectExist)
 }
 
-func (c *CLI) setupProject() string {
+func (c *CLI) setupProjectOrNamespace(project bool) string {
 	requiresTestStart()
 	newNamespace := names.SimpleNameGenerator.GenerateName(fmt.Sprintf("e2e-test-%s-", c.kubeFramework.BaseName))
-	c.SetNamespace(newNamespace).ChangeUser(fmt.Sprintf("%s-user", newNamespace))
-	framework.Logf("The user is now %q", c.Username())
+	newUsername := fmt.Sprintf("%s-user", newNamespace)
+	c.SetNamespace(newNamespace)
 
-	framework.Logf("Creating project %q", newNamespace)
-	_, err := c.ProjectClient().ProjectV1().ProjectRequests().Create(context.Background(), &projectv1.ProjectRequest{
-		ObjectMeta: metav1.ObjectMeta{Name: newNamespace},
-	}, metav1.CreateOptions{})
-	o.Expect(err).NotTo(o.HaveOccurred())
+	resource := "namespace"
+	if project {
+		resource = "project"
+	}
+
+	framework.Logf("Creating %s %q...", resource, newNamespace)
+	var nsObject *corev1.Namespace
+	if project {
+		// create project
+		c.ChangeUser(newUsername)
+		nsObject = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: newNamespace}}
+		_, err := c.ProjectClient().ProjectV1().ProjectRequests().Create(context.Background(), &projectv1.ProjectRequest{
+			ObjectMeta: metav1.ObjectMeta{Name: newNamespace},
+		}, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+	} else {
+		// create namespace
+		nsObject = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: newNamespace,
+				Annotations: map[string]string{
+					annotations.OpenShiftDescription: newUsername,
+					annotations.OpenShiftDisplayName: newNamespace,
+					"openshift.io/requester":         newUsername,
+				},
+			},
+		}
+		_, err := c.AdminKubeClient().CoreV1().Namespaces().Create(context.Background(), nsObject, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		c.kubeFramework.AddNamespacesToDelete(nsObject)
```

Comment:
Is the kubeframework wired in some way that requires us to contain it instead of creating an entirely separate thign?

Reply:
We have a few cases where we modify the internals of the test framework, and I didn't want to address that in this PR.